modules/solid_mechanics/src/materials/ConstitutiveModel.C

```diff
@@ -93,13 +95,19 @@ ConstitutiveModel::applyThermalStrain(unsigned qp,
                                       SymmTensor & strain_increment,
                                       SymmTensor & d_strain_dT)
 {
-  if (_has_temp && _t_step != 0)
+  if (_t_step >= 1)
+    _zeroth_step = false;
+
+  if (_t_step >= 2)
+    _first_step = false;
+
```

Comment:
Well would you call that? Stored copy of _t_step? Then you have a line of code that looks like `_copy_t_step = _t_step`. That's just as awkward.